test/extensions/filters/http/wasm/test_data/grpc_callout_cpp.cc

```diff
@@ -0,0 +1,51 @@
+// NOLINT(namespace-envoy)
+#include <string>
+#include <unordered_map>
+#include "proxy_wasm_intrinsics.h"
+#include "proxy_wasm_intrinsics_lite.pb.h"
+
+class ExampleContext : public Context {
+ public:
+  explicit ExampleContext(uint32_t id, RootContext *root) : Context(id, root) {}
+
+  FilterHeadersStatus onRequestHeaders() override;
+};
+
+static RegisterContextFactory
+    register_ExampleContext(CONTEXT_FACTORY(ExampleContext));
+
+class MyGrpcCallHandler : public GrpcCallHandler<google::protobuf::Value> {
+ public:
+  MyGrpcCallHandler(ExampleContext *context)
+      : GrpcCallHandler<google::protobuf::Value>(context), context_(context) {}
+  void onCreateInitialMetadata() override {}
```

Comment:
Should we convert these from pure virtuals to having a default implementation of {} so that we can avoid requiring handlers which are used?

Reply:
Sure, would you like me to do that in this PR?

Reply:
Nothing, just wondering if you thought that was a good idea.

Reply:
Oh, I'm pretty neutral on that.  Most people will copy the boilerplate anyways and then hack it up.